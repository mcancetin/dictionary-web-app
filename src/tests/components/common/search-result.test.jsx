import { render, screen } from "@testing-library/react";

import SearchResult from "@/components/common/search-result";

import { mockedWord } from "../../__mocks__";

describe("<SearchResult />", () => {
    it("should not render when data prop is falsy", () => {
        const { container } = render(<SearchResult data={null} />);
        expect(container.firstChild).toBeNull();
    });

    it("should render the search result", () => {
        const container = render(<SearchResult data={mockedWord} />);

        expect(container).toMatchSnapshot();
    });
});
