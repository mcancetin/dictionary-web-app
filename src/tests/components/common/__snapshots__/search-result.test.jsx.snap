// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SearchResult /> > should render the search result 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pt-12 dark:text-white"
      >
        <div
          class="flex items-center justify-between pb-10"
        >
          <div
            class="flex flex-col gap-2"
          >
            <span
              class="text-heading-l font-bold"
            >
              etc
            </span>
            <span
              class="text-heading-m text-vivid-purple"
            >
              /ˌɛt ˈsɛt(ə)ɹə/
            </span>
          </div>
          <button
            class="group rounded-full bg-vivid-purple/40 p-6 transition-colors duration-100 hover:bg-vivid-purple"
            type="button"
          >
            <svg
              class="lucide lucide-play fill-vivid-purple text-vivid-purple group-hover:fill-white group-hover:text-white"
              fill="none"
              height="28"
              role="img"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="6 3 20 12 6 21 6 3"
              />
            </svg>
          </button>
        </div>
        <div
          class="pb-10"
        >
          <ul
            class="space-y-10"
          >
            <li>
              <div
                class="flex items-center gap-6 pb-10"
              >
                <span
                  class="text-heading-m"
                >
                  adverb
                </span>
                <div
                  class="w-full"
                >
                  <hr
                    class="mt-1"
                  />
                </div>
              </div>
              <span
                class="block pb-6 text-heading-s text-medium-gray"
              >
                Meaning
              </span>
              <ul
                class="list-inside list-disc"
              >
                <li
                  class="mark pb-4 pl-6 marker:text-vivid-purple"
                >
                  Alternative form of etc.
                </li>
              </ul>
            </li>
          </ul>
        </div>
        <div
          class="flex items-center gap-4 border-t border-very-light-gray py-5"
        >
          <span
            class="block text-sm text-medium-gray"
          >
            Source
          </span>
          <p>
            <a
              class="pr-2 text-sm text-vivid-purple underline"
              href="https://en.wiktionary.org/wiki/etc"
              rel="noreferrer"
              target="_blank"
            >
              https://en.wiktionary.org/wiki/etc
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pt-12 dark:text-white"
    >
      <div
        class="flex items-center justify-between pb-10"
      >
        <div
          class="flex flex-col gap-2"
        >
          <span
            class="text-heading-l font-bold"
          >
            etc
          </span>
          <span
            class="text-heading-m text-vivid-purple"
          >
            /ˌɛt ˈsɛt(ə)ɹə/
          </span>
        </div>
        <button
          class="group rounded-full bg-vivid-purple/40 p-6 transition-colors duration-100 hover:bg-vivid-purple"
          type="button"
        >
          <svg
            class="lucide lucide-play fill-vivid-purple text-vivid-purple group-hover:fill-white group-hover:text-white"
            fill="none"
            height="28"
            role="img"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="6 3 20 12 6 21 6 3"
            />
          </svg>
        </button>
      </div>
      <div
        class="pb-10"
      >
        <ul
          class="space-y-10"
        >
          <li>
            <div
              class="flex items-center gap-6 pb-10"
            >
              <span
                class="text-heading-m"
              >
                adverb
              </span>
              <div
                class="w-full"
              >
                <hr
                  class="mt-1"
                />
              </div>
            </div>
            <span
              class="block pb-6 text-heading-s text-medium-gray"
            >
              Meaning
            </span>
            <ul
              class="list-inside list-disc"
            >
              <li
                class="mark pb-4 pl-6 marker:text-vivid-purple"
              >
                Alternative form of etc.
              </li>
            </ul>
          </li>
        </ul>
      </div>
      <div
        class="flex items-center gap-4 border-t border-very-light-gray py-5"
      >
        <span
          class="block text-sm text-medium-gray"
        >
          Source
        </span>
        <p>
          <a
            class="pr-2 text-sm text-vivid-purple underline"
            href="https://en.wiktionary.org/wiki/etc"
            rel="noreferrer"
            target="_blank"
          >
            https://en.wiktionary.org/wiki/etc
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
